Standalone unit test failures in Roslyn.Compilers.CompilerServer.Unittests.dll

Looking at 11 test failures. Some of them fail due to internal compiler errors where they may be missing quotations. Others aren't receiving expected error BC30037 or CS1056. 

Full failed test log below:

```
xUnit.net Console Runner (64-bit .NET 4.0.30319.42000)
  Discovering: Roslyn.Compilers.CompilerServer.UnitTests
  Discovered:  Roslyn.Compilers.CompilerServer.UnitTests
  Starting:    Roslyn.Compilers.CompilerServer.UnitTests
    Microsoft.CodeAnalysis.CompilerServer.UnitTests.EndToEndDeterminismTest.AnonType [FAIL]
      Deterministic compile failed 
 stderr: 'C:\Program' is not recognized as an internal or external command,
      operable program or batch file.
       
 stdout:  
      Expected: False
      Actual:   True
      Stack Trace:
           at Microsoft.CodeAnalysis.CompilerServer.UnitTests.EndToEndDeterminismTest.CompileAndGetBytes(String source, String additionalFlags, String& finalFlags)
           at Microsoft.CodeAnalysis.CompilerServer.UnitTests.EndToEndDeterminismTest.RunDeterministicTest(String source, String additionalFlags)
    Microsoft.CodeAnalysis.CompilerServer.UnitTests.EndToEndDeterminismTest.HelloWorld [FAIL]
      Deterministic compile failed 
 stderr: 'C:\Program' is not recognized as an internal or external command,
      operable program or batch file.
       
 stdout:  
      Expected: False
      Actual:   True
      Stack Trace:
           at Microsoft.CodeAnalysis.CompilerServer.UnitTests.EndToEndDeterminismTest.CompileAndGetBytes(String source, String additionalFlags, String& finalFlags)
           at Microsoft.CodeAnalysis.CompilerServer.UnitTests.EndToEndDeterminismTest.RunDeterministicTest(String source, String additionalFlags)
    Microsoft.CodeAnalysis.CompilerServer.UnitTests.EndToEndDeterminismTest.CallerInfo [FAIL]
      Deterministic compile failed 
 stderr: 'C:\Program' is not recognized as an internal or external command,
      operable program or batch file.
       
 stdout:  
      Expected: False
      Actual:   True
      Stack Trace:
           at Microsoft.CodeAnalysis.CompilerServer.UnitTests.EndToEndDeterminismTest.CompileAndGetBytes(String source, String additionalFlags, String& finalFlags)
           at Microsoft.CodeAnalysis.CompilerServer.UnitTests.EndToEndDeterminismTest.RunDeterministicTest(String source, String additionalFlags)
    Microsoft.CodeAnalysis.CompilerServer.UnitTests.EndToEndDeterminismTest.LineDirective [FAIL]
      Deterministic compile failed 
 stderr: 'C:\Program' is not recognized as an internal or external command,
      operable program or batch file.
       
 stdout:  
      Expected: False
      Actual:   True
      Stack Trace:
           at Microsoft.CodeAnalysis.CompilerServer.UnitTests.EndToEndDeterminismTest.CompileAndGetBytes(String source, String additionalFlags, String& finalFlags)
           at Microsoft.CodeAnalysis.CompilerServer.UnitTests.EndToEndDeterminismTest.RunDeterministicTest(String source, String additionalFlags)
    Microsoft.CodeAnalysis.CompilerServer.UnitTests.DesktopBuildClientTests+ServerTests.OnlyStartsOneServer [FAIL]
      System.Exception : Exception of type 'System.Exception' was thrown.
      Stack Trace:
           at Microsoft.CodeAnalysis.CompilerServer.UnitTests.DesktopBuildClientTests.ServerTests.<>c__DisplayClass11_0.<OnlyStartsOneServer>b__0(String[] <arguments>, BuildPaths <buildPaths>, TextWriter <textWriter>, IAnalyzerAssemblyLoader <analyzerAssemblyLoader>)
           at Microsoft.CodeAnalysis.CommandLine.DesktopBuildClient.RunLocalCompilation(String[] arguments, BuildPaths buildPaths, TextWriter textWriter)
           at Microsoft.CodeAnalysis.CommandLine.BuildClient.RunCompilation(IEnumerable`1 originalArguments, BuildPaths buildPaths, TextWriter textWriter)
           at Microsoft.CodeAnalysis.CompilerServer.UnitTests.DesktopBuildClientTests.ServerTests.OnlyStartsOneServer()
    Microsoft.CodeAnalysis.CompilerServer.UnitTests.CompilerServerUnitTests.Utf8OutputInRspFileVbc [FAIL]
      Assert.Equal() Failure
                 (pos 0)
      Expected: src.vb(1) : error BC30037: Character is núúú
      Actual:   
                 (pos 0)
      Stack Trace:
           at Microsoft.CodeAnalysis.CompilerServer.UnitTests.CompilerServerUnitTests.<Utf8OutputInRspFileVbc>d__68.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    Microsoft.CodeAnalysis.CompilerServer.UnitTests.CompilerServerUnitTests.Bug1024619_01 [FAIL]
      Assert.Equal() Failure
      Expected: 0
      Actual:   1
      Stack Trace:
           at Microsoft.CodeAnalysis.CompilerServer.UnitTests.CompilerServerUnitTests.<Bug1024619_01>d__74.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    Microsoft.CodeAnalysis.CompilerServer.UnitTests.CompilerServerUnitTests.VbcFallbackUtf8 [FAIL]
      Assert.Equal() Failure
                 (pos 0)
      Expected: test.vb(1) : error BC30037: Character is úúú
      Actual:   
                 (pos 0)
      Stack Trace:
           at Microsoft.CodeAnalysis.CompilerServer.UnitTests.CompilerServerUnitTests.<VbcFallbackUtf8>d__39.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    Microsoft.CodeAnalysis.CompilerServer.UnitTests.CompilerServerUnitTests.Utf8OutputInRspFileCsc [FAIL]
      Assert.Equal() Failure
                 (pos 0)
      Expected: src.cs(1,1): error CS1056: Unexpected chaúúú
      Actual:   
                 (pos 0)
      Stack Trace:
           at Microsoft.CodeAnalysis.CompilerServer.UnitTests.CompilerServerUnitTests.<Utf8OutputInRspFileCsc>d__67.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    Microsoft.CodeAnalysis.CompilerServer.UnitTests.CompilerServerUnitTests.Bug1024619_02 [FAIL]
      Assert.Equal() Failure
      Expected: 0
      Actual:   1
      Stack Trace:
           at Microsoft.CodeAnalysis.CompilerServer.UnitTests.CompilerServerUnitTests.<Bug1024619_02>d__75.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    Microsoft.CodeAnalysis.CompilerServer.UnitTests.CompilerServerUnitTests.ServerRespectsAppConfig [SKIP]
      DevDiv 1095079
    Microsoft.CodeAnalysis.CompilerServer.UnitTests.CompilerServerUnitTests.CscFallBackOutputUtf8 [FAIL]
      Assert.Equal() Failure
                 (pos 0)
      Expected: test.cs(1,1): error CS1056: Unexpected chúúú
      Actual:   
                 (pos 0)
      Stack Trace:
           at Microsoft.CodeAnalysis.CompilerServer.UnitTests.CompilerServerUnitTests.<CscFallBackOutputUtf8>d__37.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
  Finished:    Roslyn.Compilers.CompilerServer.UnitTests
=== TEST EXECUTION SUMMARY ===
   Roslyn.Compilers.CompilerServer.UnitTests  Total: 68, Errors: 0, Failed: 11, Skipped: 1, Time: 20.501s
```

--- a/src/Compilers/Server/VBCSCompilerTests/EndToEndDeterminismTest.cs
+++ b/src/Compilers/Server/VBCSCompilerTests/EndToEndDeterminismTest.cs
@@ -33,7 +33,7 @@ namespace Microsoft.CodeAnalysis.CompilerServer.UnitTests
                 var errorsFile = srcFile + ".errors";
 
                 // Compile
-                var result = ProcessUtilities.Run("cmd", $"/C {CompilerServerUnitTests.CSharpCompilerClientExecutable} { finalFlags } { srcFile } /out:{ outFile } > { errorsFile }");
+                var result = ProcessUtilities.Run("cmd", $@"/C ""{CompilerServerUnitTests.CSharpCompilerClientExecutable}"" { finalFlags } { srcFile } /out:{ outFile } > { errorsFile }");
                 if (result.ExitCode != 0)
                 {
                     var errors = File.ReadAllText(errorsFile);

